exercises/saddle-points/example.rs

```diff
@@ -1,23 +1,24 @@
 pub fn find_saddle_points(input: &[Vec<u64>]) -> Vec<(usize, usize)> {
-    let mut saddle_points = Vec::new();
+    let (width, height) = match (input.len(), input[0].len()) {
+        (0, _) | (_, 0) => return Vec::new(),
+        (n, k) => (n, k),
+    };
```

Comment:
This is a bit too clever: it sacrifices some clarity for the sake of brevity, and isn't faster than the straightforward implementation:

```rust
let width = input.len();
if width == 0 {
    return Vec::new();
}
let height = input[0].len();
if height == 0 {
    return Vec::new();
}
```

Reply:
Yes, my suggestion is less clear, especially for people who come from other languages

Frankly speaking we don't need `width` here at all.

```rust
    let height = if !matrix.is_empty() && !matrix[0].is_empty() {
        matrix[0].len()
    } else {
        return Vec::new();
    };

// calculate column minimums and store them in a Vec

for (i, row) in matrix.iter().enumerate() {
    let row_max = *row.iter().max().unrwap();
    
    for j in 0..height {
        // ...
    }
}
```

Reply:
Good point. I like that approach better.